src/devices/cpu/avr8/avr8.cpp

```diff
@@ -730,44 +840,179 @@ atmega2560_device::atmega2560_device(const machine_config &mconfig, const char *
+	m_timer1_ticks[(WGM1_FAST_PWM_ICR << 4) | 15] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_icr_set_set;
+
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  0] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_norm_norm;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  1] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_norm_toggle;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  2] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_norm_clear;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  3] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_norm_set;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  4] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_toggle_norm;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  5] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_toggle_toggle;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  6] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_toggle_clear;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  7] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_toggle_set;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  8] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_clear_norm;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) |  9] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_clear_toggle;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) | 10] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_clear_clear;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) | 11] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_clear_set;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) | 12] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_set_norm;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) | 13] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_set_toggle;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) | 14] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_set_clear;
+	m_timer1_ticks[(WGM1_FAST_PWM_OCR << 4) | 15] = &avr8_device<NumTimers>::timer1_tick_fast_pwm_ocr_set_set;
+
+	m_timer2_ticks[WGM02_NORMAL] = &avr8_device<NumTimers>::timer2_tick_norm;
+	m_timer2_ticks[WGM02_PWM_PC] = &avr8_device<NumTimers>::timer2_tick_default;
+	m_timer2_ticks[WGM02_CTC_CMP] = &avr8_device<NumTimers>::timer2_tick_default;
+	m_timer2_ticks[WGM02_FAST_PWM] = &avr8_device<NumTimers>::timer2_tick_fast_pwm;
+	m_timer2_ticks[WGM02_RESERVED0] = &avr8_device<NumTimers>::timer2_tick_default;
+	m_timer2_ticks[WGM02_PWM_PC_CMP] = &avr8_device<NumTimers>::timer2_tick_default;
+	m_timer2_ticks[WGM02_RESERVED1] = &avr8_device<NumTimers>::timer2_tick_default;
+	m_timer2_ticks[WGM02_FAST_PWM_CMP] = &avr8_device<NumTimers>::timer2_tick_fast_pwm_cmp;
+
+	m_timer0_tick = m_timer0_ticks[0];
+	m_timer1_tick = m_timer1_ticks[0];
+	m_timer2_tick = m_timer2_ticks[0];
 }
 
+//-------------------------------------------------
+//  avr8_device - constructor
+//-------------------------------------------------
+template class avr8_device<2>;
+template class avr8_device<3>;
+template class avr8_device<6>;
```

Comment:
Because compilers can be stupid sometimes, you probably need to move the explicit instantiations to the bottom of the file to fix the linker errors.

Reply:
Moved to the bottom of the file as suggested.